hoodie-client/src/main/java/com/uber/hoodie/HoodieWriteClient.java

```diff
@@ -772,107 +775,176 @@ public boolean rollbackToSavepoint(String savepointTime) {
    * files. (4) Finally delete .commit or .inflight file,
    */
   public boolean rollback(final String commitTime) throws HoodieRollbackException {
-    rollback(Lists.newArrayList(commitTime));
+    rollbackInternal(commitTime);
     return true;
   }
 
-  private void rollback(List<String> commits) {
-    if (commits.isEmpty()) {
-      logger.info("List of commits to rollback is empty");
-      return;
+  /**
+   * NOTE : This action requires all writers (ingest and compact) to a dataset to be stopped before proceeding.
+   * Revert the (inflight/committed) record changes for all commits after the provided @param.
+   * Three steps: (1) Atomically unpublish this commit (2) clean indexing data, (3) clean new generated parquet/log
+   * files and/or append rollback to existing log files. (4) Finally delete .commit, .inflight, .compaction.inflight
+   * or .compaction.requested file
+   */
+  public void restoreToCommit(final String commitTime) throws HoodieRollbackException {
```

Comment:
rename arg from `commitTime` to `instantTime`, rename method to `restoreToInstant`

(this file has remanants of old naming from the past e.g referring to instantTime as commitTime and instant as commit generically..  Cleanup would be appreciated. )

Reply:
Also can you move all the Restore related logic to a new class, leaving the WriteClient uncluttered.. We can later figure out how we can create a write client out of it.. `rollback()` here simply needs to be on inflight instants... Makes sense? Goes back to the original discussion you raised on the PR as well

Reply:
Cleaned up `commit` references to `instants`. Moving the restore logic to a separate class is tricky since there are common internal methods used between `restoreToInstant()` and `rollback()`. To do that, we'd have to duplicate some code or share internal private methods, both of which are ugly. Let's chat f2f